app/controllers/jobs_controller.rb

```diff
@@ -15,7 +17,23 @@ def create
     end
   end
 
+  def show
+    @user = current_user
+    @job = Job.find(params[:id])
+  end
+
   private
+    def validate_authorization
+      unless authorized?
+        render file: 'public/404.html', status: :not_found, layout: false
+      end
+    end
+
+    def authorized?
+      current_user.professional? ||
+      current_user.jobs.pluck(:id).include?(params[:id].to_i)
```

Comment:
I like this method a lot. You are authorized if you are a professional or if the job belongs to you.